flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSource.java

```diff
@@ -409,6 +415,9 @@ public int hashCode() {
                         OffsetsInitializer.timestamp(startupTimestampMillis));
                 break;
         }
+        if (setBounded != null) {
+            kafkaSourceBuilder.setBounded(OffsetsInitializer.offsets(setBounded));
```

Comment:
The `OffsetsInitializer.offsets(setBounded)` is the `OffsetResetStrategy.EARLIEST` strategy, which couldn't be set as stoppingOffsetsInitializer.